### PR TITLE
Tab width

### PR DESCRIPTION
**Pull request by lievenhey**

### CODE REVIEW

File: tests/modeltests/tst_formatting.cpp [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 #include "../testutils.h"
 
+#include <hotspot-config.h>
+
 #if KFSyntaxHighlighting_FOUND
```
Comment (milianw) [inline tests/modeltests/tst_formatting.cpp:18]: was this silently causing compiler warnings? warnings should ideally make the CI non-green (do we need -Werror?)

File: src/models/highlightedtext.h [hunk 2]
```diff
@@ -29,7 +27,7 @@ class HighlightedText : public QObject
 {
     Q_OBJECT
 public:
-    HighlightedText(KSyntaxHighlighting::Repository* repository, QObject* parent = nullptr);
+    HighlightedText(KSyntaxHighlighting::Repository* repository, int tabWidth, QObject* parent = nullptr);
```
Comment (milianw) [inline src/models/highlightedtext.h:30]: why is this passed as a property here, can't we just call updateTabWidth as needed?

File: src/models/highlightedtext.cpp [hunk 3]
```diff
@@ -276,16 +292,17 @@ class HighlightedLine
     HighlightingImplementation* m_highlighter = nullptr;
     QString m_text;
     int m_index = -1;
+    int m_tabWidth = 8;
```
Comment (milianw) [inline src/models/highlightedtext.cpp:295]: default should probably be set to 4?

File: src/models/sourcecodemodel.h [hunk 4]
```diff
@@ -28,7 +28,7 @@ class SourceCodeModel : public QAbstractTableModel
 {
     Q_OBJECT
 public:
-    explicit SourceCodeModel(KSyntaxHighlighting::Repository* repository, QObject* parent = nullptr);
+    explicit SourceCodeModel(KSyntaxHighlighting::Repository* repository, int tabWidth = 8, QObject* parent = nullptr);
```
Comment (milianw) [inline src/models/sourcecodemodel.h:31]: repeated value of 8 - this is uncommon, no? 4 is much more often used I believe

File: src/resultsdisassemblypage.cpp [hunk 5]
```diff
@@ -221,11 +221,11 @@ ResultsDisassemblyPage::ResultsDisassemblyPage(CostContextMenu* costContextMenu,
     , ui(std::make_unique<Ui::ResultsDisassemblyPage>())
 #if KFSyntaxHighlighting_FOUND
     , m_repository(std::make_unique<KSyntaxHighlighting::Repository>())
-    , m_disassemblyModel(new DisassemblyModel(m_repository.get(), this))
-    , m_sourceCodeModel(new SourceCodeModel(m_repository.get(), this))
+    , m_disassemblyModel(new DisassemblyModel(m_repository.get(), 8, this))
```
Comment (milianw) [inline src/resultsdisassemblypage.cpp:224]: here too - please introduce a central `DefaultTabWidth` somewhere and use it everywhere

File: src/disassemblysettingspage.ui [hunk 6]
```diff
@@ -32,14 +32,14 @@
    <property name="bottomMargin">
```
Comment (milianw) [inline src/disassemblysettingspage.ui:32]: commenting here as github still does not allow me to comment on the commit message:

you have a typo (with -> width), and you are missing some explanatory text. please write a short summary of how things are being tied together

**Review by milianw — CHANGES_REQUESTED** [review]

File: tests/modeltests/tst_formatting.cpp (hunk 1 above)
Comment (lievenhey) [reply tests/modeltests/tst_formatting.cpp:18]: oh no, it just didn't run the ksyntaxhighlighting tests

File: src/models/highlightedtext.cpp [hunk 7]
```diff
@@ -250,13 +253,24 @@ class HighlightedLine
         m_layout = nullptr;
     }
 
+    void setTabWidthInPixels(int tabWidthInPixels)
+    {
+        m_tabWidthInPixels = tabWidthInPixels;
+        m_layout = nullptr;
```
Comment (milianw) [inline src/models/highlightedtext.cpp:259]: `.reset()` is more expressive

File: tests/modeltests/tst_formatting.cpp (hunk 1 above)
Comment (milianw) [reply tests/modeltests/tst_formatting.cpp:18]: I still don't understand this. it should have triggered a compiler warning before. why didn't it?

**Review by milianw — APPROVED** [review]

**Review by milianw — APPROVED** [review]

**Pull request merged**